gtsam/inference/EliminateableFactorGraph.h

```diff
@@ -194,37 +213,63 @@ namespace gtsam {
+     *         i.e. all variables not in \c variables.
      *  @param function Optional dense elimination function, if not provided the default will be
      *         used.
      *  @param variableIndex Optional pre-computed VariableIndex for the factor graph, if not
      *         provided one will be computed. */
     boost::shared_ptr<BayesNetType> marginalMultifrontalBayesNet(
       boost::variant<const Ordering&, const KeyVector&> variables,
-      OptionalOrdering marginalizedVariableOrdering = boost::none,
+      const Ordering& marginalizedVariableOrdering,
+      const Eliminate& function = EliminationTraitsType::DefaultEliminate,
+      OptionalVariableIndex variableIndex = boost::none) const;
+
+    /** Compute the marginal of the requested variables and return the result as a Bayes tree.  Uses
+     *  COLAMD marginalization order by default
+     *  @param variables Determines the variables whose marginal to compute, if provided as an
+     *         Ordering they will be ordered in the returned BayesNet as specified, and if provided
+     *         as a KeyVector they will be ordered using constrained COLAMD.
+     *  @param function Optional dense elimination function, if not provided the default will be
+     *         used.
+     *  @param variableIndex Optional pre-computed VariableIndex for the factor graph, if not
+     *         provided one will be computed. */
+    boost::shared_ptr<BayesTreeType> marginalMultifrontalBayesTree(
+      boost::variant<const Ordering&, const KeyVector&> variables,
       const Eliminate& function = EliminationTraitsType::DefaultEliminate,
       OptionalVariableIndex variableIndex = boost::none) const;
 
     /** Compute the marginal of the requested variables and return the result as a Bayes tree.
      *  @param variables Determines the variables whose marginal to compute, if provided as an
      *         Ordering they will be ordered in the returned BayesNet as specified, and if provided
      *         as a KeyVector they will be ordered using constrained COLAMD.
-     *  @param marginalizedVariableOrdering Optional ordering for the variables being marginalized
-     *         out, i.e. all variables not in \c variables.  If this is boost::none, the ordering
-     *         will be computed with COLAMD.
+     *  @param marginalizedVariableOrdering Ordering for the variables being marginalized out,
+     *         i.e. all variables not in \c variables.
      *  @param function Optional dense elimination function, if not provided the default will be
      *         used.
      *  @param variableIndex Optional pre-computed VariableIndex for the factor graph, if not
      *         provided one will be computed. */
     boost::shared_ptr<BayesTreeType> marginalMultifrontalBayesTree(
       boost::variant<const Ordering&, const KeyVector&> variables,
-      OptionalOrdering marginalizedVariableOrdering = boost::none,
+      const Ordering& marginalizedVariableOrdering,
```

Comment:
Since this `OptionalOrdering` is in the middle of the args list we would expect people to have a slight change of their code (remove a wandering `boost::none`) @dellaert

Reply:
Also no change is necessary @gchenfc, this is just a marker for blocking :)